packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -27,6 +27,28 @@ export type ButtonProps = {
   touchRippleRef?: any; // FIXME(v8:romgrk): find a way to remove
 };
 
+export type CheckboxProps = {
+  ref?: Ref<HTMLButtonElement>;
+  id?: string;
+  checked?: boolean;
+  className?: string;
+  disabled?: boolean;
+  indeterminate?: boolean;
+  inputRef?: React.Ref<HTMLInputElement>;
+  name?: string;
+  label?: React.ReactNode;
+  onClick?: React.MouseEventHandler;
+  onChange?: React.ChangeEventHandler;
+  onKeyDown?: React.KeyboardEventHandler;
+  size?: 'small' | 'medium';
+  slotProps?: {
+    htmlInput?: React.InputHTMLAttributes<HTMLInputElement>;
+  };
+  style?: React.CSSProperties;
+  tabIndex?: number;
+  touchRippleRef?: any; // FIXME(v8:romgrk): find a way to remove
```

Comment:
How about moving this part to `BaseCheckbox` and passing `hasFocus` prop to it?

https://github.com/mui/mui-x/blob/5ed0b653461aa4dbbc1ddfd606310e396b77465c/packages/x-data-grid/src/components/columnSelection/GridCellCheckboxRenderer.tsx#L72-L80

Reply:
Yeah that could work. I wonder if `autoFocus` would be better? It would match the similar prop on `MenuList`.

Reply:
> I wonder if autoFocus would be better?

Yeah, this makes sense 👍🏻